sktime/forecasting/trend/_curve_fit_forecaster.py

```diff
@@ -0,0 +1,191 @@
+# !/usr/bin/env python3 -u
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Implements FunctionForecaster."""
+
+__author__ = ["benheid"]
+__all__ = ["CurveFitForecaster"]
+
+import pandas as pd
+from scipy.optimize import curve_fit
+
+from sktime.forecasting.base import BaseForecaster
+from sktime.forecasting.trend._util import _get_X_numpy_int_from_pandas
+
+
+class CurveFitForecaster(BaseForecaster):
+    """The CurveFitForecaster takes a function and fits it by using scipy curve_fit.
+
+    The CurveFitForecaster applies the scipy curve_fit method to find the best
+    parameter of a function. This function maps a list of integers to real values.
+    The integers are derrived from the index of the input time series.
+
+    In `fit`
+    1. The index of the input time series is transformed to a list of Integers.
+    2. The scipy curve_fit is called using the list of integers as x values,
+       and the time series values as y values.
+    In `predict`
+    1. The ForecastingHorizon is transformed to a list of integers.
+    2. The list of integers is passed together with the fitted parameters to the
+       function to provide the forecast.
+
+
+    Parameters
+    ----------
+    function: Callable[[Iterable[int], ...], Iterable[float]]
+        The function that should be fitted and used to make forecasts.
+        The signature of the functions is `function(x, ...)`.
+        It takes the independet variables as first argument and the parametrs
+        to fit as separate remaining arguments.
+        See scipy.optimize.curve_fit for more information.
+    curve_fit_params: dict, default=None
+        Additional parameters that should be passed to the curve_fit method.
+        See scipy.optimize.curve_fit for more information.
+
```

Comment:
should be only one newline, but there are two

Reply:
Thank you, I removed one line